### ISSUE TITLE
Token renew only works if endpoints are passed to config, even thought it's marked optional

### ISSUE THREAD

**Issue by Valchris** [AzureAD/azure-activedirectory-library-for-js#195]
I believe the offending code lies in ProtectedResourceInteceptor, the else if condition should actually be:
```javascript
} else if (authService.config && (!authService.config.endpoints || isEndpoint)) {
```
Snippet below:
```javascript
if (authService.loginInProgress()) {
    authService.info('login already start.');
    return $q.reject();
} else if (authService.config && isEndpoint) {
    // external endpoints
    // delayed request to return after iframe completes
    var delayedRequest = $q.defer();
    authService.acquireToken(resource).then(function (token) {
    authService.verbose('Token is avaliable');
    config.headers.Authorization = 'Bearer ' + token;
    delayedRequest.resolve(config);
}, function (err) {
    delayedRequest.reject(err);
});
```

**Comment by AndreSayreMSFT** [comment]
@weijjia We seem to be hitting this, or at least a related problem, internally!

The existence of this issue prevents the Authorization token from being attached to the first of our outgoing $http calls, thus causing all sorts of badness for us.

CC: @yanchenw

**Comment by tushargupta51** [comment]
Fixed in #240.
